src/core/pressure.cpp

```diff
@@ -123,7 +123,10 @@ void pressure_calc(double *result, double *result_t, double *result_nb, double *
   calc_long_range_virials();
 
 #ifdef VIRTUAL_SITES_RELATIVE  
-  vs_relative_pressure_and_stress_tensor(virials.vs_relative,p_tensor.vs_relative);
+  std::shared_ptr<VirtualSitesRelative> vsr =std::dynamic_pointer_cast<VirtualSitesRelative>(virtual_sites());
```

Comment:
I don't like this as it breaks dependency inversion. The client should not have to know about specific implementations of VS. Please add this as virtual function to the VS class, with a default implementation that does either nothing or throws an error/warning as you see fit. While you're on it you can also replace the output params and return by value.